src/components/Carousel.tsx

```diff
@@ -1,18 +1,113 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+const GAP_IMG = 10;
+
+type Props = {
+  images: string[],
+  step: number,
+  frameSize: number,
+  itemWidth: number,
+  animationDuration: number,
+  infinite: boolean;
+};
+
+const Carousel: React.FC<Props> = ({
+  images,
+  step,
+  frameSize,
+  itemWidth,
+  animationDuration,
+  infinite,
+}) => {
+  const [startImage, setStartImage] = useState(0);
+  const endIndexImage = startImage + frameSize;
+
+  const widthContainer = frameSize * itemWidth + GAP_IMG * frameSize;
+  let translateXValue = startImage * (itemWidth + GAP_IMG);
+
+  if (startImage === 0) {
+    translateXValue = 0;
+  }
```

Comment:
do not forget about ternary
```suggestion
  const translateXValue = startImage === 0 ? 0 : startImage * (itemWidth + GAP_IMG);
```